plugins/catalog-backend/src/next/DefaultLocationService.ts

```diff
@@ -41,30 +42,45 @@ export class DefaultLocationService implements LocationService {
         apiVersion: 'backstage.io/v1alpha1',
         kind: 'Location',
         metadata: {
-          name: `${spec.type}:${spec.target}`,
+          name: locationSpecToMetadataName({
+            type: spec.type,
+            target: spec.target,
+          }),
           namespace: 'default',
           annotations: {
             [LOCATION_ANNOTATION]: `${spec.type}:${spec.target}`,
             [ORIGIN_LOCATION_ANNOTATION]: `${spec.type}:${spec.target}`,
           },
         },
         spec: {
-          location: { type: spec.type, target: spec.target },
+          type: spec.type,
+          target: spec.target,
         },
       };
-      const processed = await this.orchestrator.process({
-        entity,
-        eager: true,
-        state: new Map(),
-      });
-      if (processed.ok) {
-        return {
-          location: { ...spec, id: `${spec.type}:${spec.target}` },
-          entities: [processed.completedEntity],
-        };
+      const unprocessedEntities: Entity[] = [entity];
+      const entities: Entity[] = [];
+      const state = new Map(); // ignored
+      while (unprocessedEntities.length) {
+        const currentEntity = unprocessedEntities.pop();
+        if (!currentEntity) {
```

Comment:
I mean it shouldn't, but if one of the unprocessed entities is `null` or `undefined` then it would get added to the array.

Typescript complains otherwise I guess..